libkbfs/md_journal.go

```diff
@@ -651,7 +667,7 @@ func (j *mdJournal) convertToBranch(
 		j.log.CDebugf(ctx, "Old prev root of rev=%s is %s",
 			brmd.RevisionNumber(), brmd.GetPrevRoot())
 
-		if i > 0 {
+		if len(mdsToRemove) > 0 {
```

Comment:
This could use a comment now, I think. Something like "Set the prev root for everything after the first MD we modify, which happens to be indicated by mdsToRemove being non-empty."